src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/StreamHelper.cs

```diff
@@ -259,6 +272,25 @@ private void Initialize()
                 throw;
             }
         }
+
+        private enum FileSizeUnit
+        {
+            Byte, KB, MB, GB, TB, PB, EB
+        }
+
+        internal static string DisplayHumanReadableFileSize(long bytes)
```

Comment:
I suggest to move this in Utils file - we could use this another places (Formatting System).
After that we can remove adding InternalsVisibleTo from the PR.

Also I think we need optimize the method because it will be called frequently.

Reply:
OK, but the `InternalsVisibleTo` is to allow for testing in the Xunit project to avoid making it public. Since `Invoke-WebRequest`'s progress reporting was already throttled in PowerShell Core, I don't think this method gets called too often but can check

Reply:
Utils file is for SMA and SMA already has the attribute.

Reply:
Are you sure around moving this internal method to [Utils.cs](https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/engine/Utils.cs) in SMA? Because I thought @SteveL-MSFT wanted to separate the `Microsoft.PowerShell.Utility` cmdlets like `Invoke-WebRequest` into its own module in the future, which would require the cmdlets to be self-contained and not depend on non-public methods in SMA.

I did a test and the method gets called around 300 times a second. Whilst one could throttle the calls to the existing calls to `WriteProgress` API in the first place in the cmdlet, I am not sure if it makes a measurable difference. To give some figures: I called this function in a dummy console app a million times with random values and it only took around 800ms, then I simplified the code to take out the expensive operations like `Math.Log10`, `Math.Pow` or floating point operations and I could not measure a time difference...

Reply:
> in the future, which would require the cmdlets to be self-contained and not depend on non-public methods in SMA.

When the time comes we will decide it. We must now follow the current order. Please move the method.

>  I called this function in a dummy console app a million times with random values and it only took around 800ms, then I simplified the code to take out the expensive operations like Math.Log10, Math.Pow or floating point operations and I could not measure a time difference...

I guess your tests/implementation was not so good. Obviously Math and floating operation is significant slow.

Maybe I'm too careful but we've already fixed performance issues in progress bar and I wouldn't want regression.

Reply:
Moved now into Utils.cs. Let's continue performance discussion in other comment [here](https://github.com/PowerShell/PowerShell/pull/14611#discussion_r557845451). Personally, I think we are comparing the progress bar issue that dominated the performance with something that less than a 10th of a per-cent.